lib/src/Base/Func/PiecewiseHermiteEvaluation.cxx

```diff
@@ -113,34 +153,58 @@ Point PiecewiseHermiteEvaluation::operator () (const Point & inP) const
-    iRight = iLeft + 1;
-  }
   else
-    // Find the segment containing x by bisection
-    while (iRight - iLeft > 1)
-    {
-      const UnsignedInteger im = (iRight + iLeft) / 2;
-      if (x < locations_[im]) iRight = im;
-      else iLeft = im;
-    }
+    iLeft = findSegmentIndexHermite(locations_, x, 0);
 
-  const Scalar h = locations_[iRight] - locations_[iLeft];
+  const Scalar h = locations_[iLeft + 1] - locations_[iLeft];
   const Scalar theta = (x - locations_[iLeft]) / h;
-  const Point vLeft(values_[iLeft]);
-  const Point vRight(values_[iRight]);
-  const Point dvLeft(derivatives_[iLeft]);
-  const Point dvRight(derivatives_[iRight]);
   const UnsignedInteger dimension = getOutputDimension();
   Point value(dimension);
   const Scalar alpha = 1.0 - theta;
   const Scalar beta = theta * alpha;
   const Scalar gamma = 2.0 * theta - 1.0;
-  for (UnsignedInteger i = 0; i < dimension; ++i) value[i] = alpha * vLeft[i] + theta * vRight[i] + beta * (gamma * (vRight[i] - vLeft[i]) + h * (alpha * dvLeft[i] - theta * dvRight[i]));
+  for (UnsignedInteger i = 0; i < dimension; ++i) value[i] = alpha * values_(iLeft, i) + theta * values_(iLeft + 1, i) + beta * (gamma * (values_(iLeft + 1, i) - values_(iLeft, i)) + h * (alpha * derivatives_(iLeft, i) - theta * derivatives_(iLeft + 1, i)));
   return value;
 }
 
+Sample PiecewiseHermiteEvaluation::operator () (const Sample & inSample) const
+{
+  if (inSample.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: expected an input sample of dimension 1, got dimension=" << inSample.getDimension();
+  const UnsignedInteger size = inSample.getSize();
+  const UnsignedInteger dimension = getOutputDimension();
+  Sample output(size, dimension);
+  const UnsignedInteger maxIndex = locations_.getSize() - 1;
+  UnsignedInteger iLeft = 0;
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    const Scalar x = inSample(i, 0);
+    if (x <= locations_[0])
+    {
+      for (UnsignedInteger j = 0; j < dimension; ++j) output(i, j) = values_(0, j);
+      continue;
+    }
+    UnsignedInteger iRight = maxIndex;
+    if (x >= locations_[iRight])
+    {
+      for (UnsignedInteger j = 0; j < dimension; ++j) output(i, j) = values_(iRight, j);
+      continue;
+    }
+    if (isRegular_)
+      iLeft = static_cast<UnsignedInteger>(floor((x - locations_[0]) / (locations_[1] - locations_[0])));
+    else
+      iLeft = findSegmentIndexHermite(locations_, x, iLeft);
+
+    const Scalar h = locations_[iLeft + 1] - locations_[iLeft];
+    const Scalar theta = (x - locations_[iLeft]) / h;
+    const Scalar alpha = 1.0 - theta;
+    const Scalar beta = theta * alpha;
+    const Scalar gamma = 2.0 * theta - 1.0;
+    for (UnsignedInteger j = 0; j < dimension; ++j) output(i, j) = alpha * values_(iLeft, i) + theta * values_(iLeft + 1, i) + beta * (gamma * (values_(iLeft + 1, i) - values_(iLeft, i)) + h * (alpha * derivatives_(iLeft, i) - theta * derivatives_(iLeft + 1, i)));
+    return output;
+  }
+}
```

Comment:
@dbarbier I didn't notice before but the return should be outside the for loop otherwise all components are not computed.